zboxcore/sdk/copyworker.go

```diff
@@ -243,4 +247,127 @@ func (req *CopyRequest) ProcessCopy() error {
 				req.Consensus.consensusThresh, req.Consensus.consensus))
 	}
 	return nil
+}
+
+type CopyOperation struct {
+	remotefilepath string
+	destPath       string
+	ctx            context.Context
+	ctxCncl        context.CancelFunc
+	copyMask       zboxutil.Uint128
+	maskMU         *sync.Mutex
+
+	Consensus
+}
+
+func (co *CopyOperation) Process(allocObj *Allocation, connectionID string, totalOperation int) ([]fileref.RefEntity, error) {
+	// make copyRequest object
+	cR := &CopyRequest{
+		allocationObj:  allocObj,
+		allocationID:   allocObj.ID,
+		allocationTx:   allocObj.Tx,
+		connectionID:   connectionID,
+		blobbers:       allocObj.Blobbers,
+		remotefilepath: co.remotefilepath,
+		destPath:       co.destPath,
+		ctx:            co.ctx,
+		ctxCncl:        co.ctxCncl,
+		copyMask:       co.copyMask,
+		maskMU:         co.maskMU,
+	}
+	cR.consensusThresh = co.consensusThresh
+	cR.fullconsensus = co.fullconsensus
+	numList := len(cR.blobbers)
+	objectTreeRefs := make([]fileref.RefEntity, numList)
+	blobberErrors := make([]error, numList)
+
+	wg := &sync.WaitGroup{}
+	var pos uint64
+
+	for i := cR.copyMask; !i.Equals64(0); i = i.And(zboxutil.NewUint128(1).Lsh(pos).Not()) {
```

Comment:
This for loop is same in ProcessRename and Process function. Instead of repeating code lets have a common function where for loop is present that both ProcessRename and Process will call.
Same for other workers.